server.js

```diff
@@ -1,120 +1,75 @@
-            fs.writeFileSync('user.json', JSON.stringify(fileData, null, 2));
-        }
-        const jsonData = require('./user.json');
-        res.status(201).json({
-            message: 'Successfully created a specific user!',
-            result: jsonData,
-        })
+      res.status(404).json({
+        message: 'Not Found',
+      })
     }
-})
+  });
 
-// get one specific user
-app.get("/users/:Id", (req, res) => {
-    const jsonData = require('./user.json');
-    const requestedObject = req.params.Id;
-    let indexGet = jsonData.findIndex(requestedObject => {
-        return requestedObject.name === `${req.params.Id}`;
-    });
-    if (indexGet > -1) {
-        res.status(200).json({
-            message: 'Successfully retrieved a specific user!',
-            result: jsonData[indexGet]
+// Validate and add users to the user.json, if the users file does not exist, creates the file with append.
+app.post("/users", (req, res) => {
+    const validation = (myUtilityFunctions.validationUserPost(req.body) == true);
+    if (validation) {
+        res.status(201).json({
+            message: `The user ${req.body.name} is succesfully created!`
         });
+        return myUtilityFunctions.addUserPost(req.body);
     } else {
-        res.status(404).json({
-            message: 'Not Found',
+        res.status(401).json ({
+            message: 'The user cannot be validated!'
         })
+        return validation;
     }
 });
 
-//now i can update any parameter of users, ot a total user if i change a little bit at line 58
+// The first if add a user and creates the file if the user.json file does not exist yet.
 app.put("/users/:Id", (req, res) => {
-    const jsonData = require('./user.json');
-    const updatedObject = req.params.Id;
-    let indexPut = jsonData.findIndex(updatedObject => {
-        return updatedObject.name === `${req.params.Id}`;
-    });
-    if (indexPut > -1) {
-        jsonData[indexPut] = req.body;
-        res.status(200).json
+    if (myUtilityFunctions.putUserbyId(req.body, req.params.Id) == true) {
+        console.log(myUtilityFunctions.putUserbyId(req.body, req.params.Id));
+        res.status(200).json({
+            message: `Successfully changed the user ${req.body.name}!`,
+        })
     } else {
         res.status(404).json({
-            message: 'Not Found',
+            message: `The user ${req.params.Id} does not exist!`,
         })
     }
-    res.send('It works!');
 });
 
-// delete a user named by Id
+// Delete a user bij Id.
 app.delete("/users/:Id", (req, res) => {
-    const jsonData = require('./user.json');
-    const tobedeletedObject = req.params.Id;
-    let indexDel = jsonData.findIndex(tobedeletedObject => {
-        return tobedeletedObject.name === `${req.params.Id}`;
-    });
-    if (indexDel > -1) {
-        jsonData.splice(indexDel, 1);
-        res.status(200).json
-        res.send(jsonData);
+    if (myUtilityFunctions.deleteUserById(req.params.Id) == true) {
```

Comment:
I like that you are returning the result of the function as a simple Boolean and then using it here.

It makes the utility function much less decoupled of the other parts of the application (no req/res etc).

This is the way 🤩